### PR TITLE
fix: use max_retries instead of max_tasks for retry limits

### PR DESCRIPTION
**Pull request by chapel**
## Summary

- `budget.max_tasks` was incorrectly used as the retry limit for transient spawn failures in `dispatch.ts` and consecutive failure thresholds in `invocation.ts`
- Added `budget.max_retries` (optional int >= 0, default 3) to `AgentBudgetSchema`
- Fixed both `dispatch.ts` and `invocation.ts` to use `max_retries` instead of `max_tasks`
- Added `--max-retries` CLI option to `meta add agent` and `meta set agent`
- Updated test budgets and comments in invocation lifecycle tests

## Test plan

- [x] Schema tests pass (33 tests)
- [x] Meta tests pass (159 tests)
- [x] Dispatch engine tests pass (55 tests)
- [x] Invocation lifecycle tests pass (37 tests)
- [x] CLI agent command tests pass (50 tests)

Task: @01KK0JFD
Spec: @agent-dispatch-engine

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/cli/commands/meta.ts [hunk 1]
```diff
@@ -1416,10 +1420,11 @@ Examples:
             // AC: @agent-definition-schema ac-12
             dispatch: dispatchRules,
             skills: options.skill || [],
-            ...(maxTasks !== undefined || timeoutMinutes !== undefined
+            ...(maxTasks !== undefined || maxRetries !== undefined || timeoutMinutes !== undefined
               ? {
                   budget: {
                     ...(maxTasks !== undefined && { max_tasks: maxTasks }),
+                    ...(maxRetries !== undefined && { max_retries: maxRetries }),
```
Comment (chapel) [inline src/cli/commands/meta.ts]: **SHOULD-FIX**: `max_retries` is now written in `meta add`/`meta set`, but this PR does not add coverage for those CLI paths. Please add `tests/meta.test.ts` cases for `--max-retries` on both commands (including `0` and invalid values) so this new behavior is regression-protected.

File: src/cli/commands/meta.ts [hunk 2]
```diff
@@ -1335,6 +1335,7 @@ Examples:
       "Auto-approve tasks without human confirmation (for agents)",
     )
     .option("--max-tasks <n>", "Maximum tasks budget (for agents)")
+    .option("--max-retries <n>", "Maximum retry attempts on failure (for agents, default 3)")
```
[the remark this answers is not in the document]
Comment (chapel) [reply src/cli/commands/meta.ts]: Fixed in 87b268b73 — added 3 tests: meta add with --max-retries, meta set with --max-retries, and zero-value edge case.
